### ISSUE TITLE
Test and improve the performance of local crawler

### ISSUE THREAD

**Issue by Jocelyn0830** [privacy-tech-lab/gpc-web-crawler#37]
As we discussed, we will switch back to the local crawler.

**Comment by Jocelyn0830** [comment]
I set up the local crawler on the two Mac Minis. The crawler crashed several times on one of the Mac Minis. I hypothesized that it was because npm and node versions were too old. I will update it and redo the testing next week. 

For the timing part, each crawler took about 2.5 hours to run 700 sites, which seem to be acceptable for me. Now the timeout allowed for each site to load is 20s. It is longer than what Kate and I set when we tested on our own Macs. We can consider shortening the time for higher efficiency after more testing on Mac Minis.

**Comment by katehausladen** [comment]
Jocelyn noticed that the mac minis performed worse than our macs in terms of success rate. We decided to modify the code so that if the extension fails to analyze a site, the crawler will try one more time. The changes are reflected in the above commit.

**Comment by Jocelyn0830** [comment]
I tested the updated crawler on two mac minis using 1860 sites. 
Updates include: 
1. Each mac mini (crawler) has its own table in the SQL database
2. The crawler analyzes a site one more time if it fails
3. Site id in the csv file is recorded in the SQL database

The success rate is 788/930, about 85%, on mac mini1. 807/930, about 87%, on mac mini2. The good news is crawlers are robust, so I did not need to babysit them. It is sufficient to just start them and wait till them finish. 

I will take a closer look at the results. Here are several things worth investigating:
1. Are the site ids accurate? That says, the site id associated with a site in the sql database should be the same site id of that site in the csv file.
2. Does allowing a failed site to be crawled again improve our success rate?
3. Is analysis data accurate? It might be helpful to spotcheck some websites by hand and compare those results with our analysis data.
4. Why are success rates lower than what we got on our Macbooks? It might be helpful to look at the console to explore why the sites failed and see if we can tackle any failure.
